connector/tests/src/test/java/org/oaebudt/edc/ManagementApiTransferTest.java

```diff
@@ -438,6 +444,77 @@ public void shouldUploadReportAndConsumerShouldConsumeReport() throws IOExceptio
 
         consumerEdrReceiver.stop();
     }
+    @Test
+    public void shouldUploadReportAndConsumerShouldAllowMultipleSameTypeReportInCatalog() throws IOException {
+        PROVIDER.setAuthorizationToken(KEYCLOAK_EXTENSION.getToken());
+        CONSUMER.setAuthorizationToken(KEYCLOAK_EXTENSION.getToken());
+        String reportUri = PROVIDER.getWebServiceUrl().get().toString() + "report/upload";
+        String participantUri = PROVIDER.getWebServiceUrl().get().toString() + "participant/group";
+        String accessToken = KEYCLOAK_EXTENSION.getToken();
+        final var consumerEdrReceiver = ClientAndServer.startClientAndServer(getFreePort());
+        consumerEdrReceiver.when(request("/edr")).respond(response());
+
+        Map<String, Object> jsonBodyMap = Map.of(
+                "groupName", "friends",
+                "participants", List.of(CONSUMER.getId())
+        );
+
+        String jsonBody = objectMapper.writeValueAsString(jsonBodyMap);
+
+        RestAssured
+                .given()
+                .header("Content-Type", "application/json")
+                .header(HttpHeaderNames.AUTHORIZATION, "Bearer " + accessToken)
+                .body(jsonBody)
+                .when()
+                .post(participantUri)
+                .then()
+                .statusCode(201);
+
+        Map<String, Object> jsonContentMap = Map.of(
+                "userId", 123,
+                "status", "active"
+        );
+
+        String jsonContent = objectMapper.writeValueAsString(jsonContentMap);
+
+        sendMultipartRequest(accessToken, jsonContent, metadata, reportUri);
+        sendMultipartRequest(accessToken, jsonContent, metadata, reportUri);
+
+
+        JsonArray jsonArray = CONSUMER.getCatalogDatasets(PROVIDER);
+
+        Pattern pattern = Pattern.compile("^TITLE_REPORT.*");
+
+        long count = jsonArray.stream()
+                .map(JsonObject.class::cast)
+                .map(obj -> obj.getString("@id", ""))
+                .filter(id -> pattern.matcher(id).matches())
+                .count();
+
+        Assertions.assertThat(count > 1).isTrue();
```

Comment:
`assertThat(count).isGreaterThan(1)`

but also all the logic used to extract `count` can be translated in assertJ, like
```
assertThat(jsonArray)
    .map(JsonObject.class::cast)
    .map(obj -> obj.getString("@id", ""))
    .anyMatch(it -> it.startsWith("TITLE_REPORT"));
```

Reply:
yes, but it's more readable :)